tests/integration/lifecycle/test_lifecycle_organize_to_overlay.py

```diff
@@ -0,0 +1,279 @@
+            "foo",
+            Step.OVERLAY,
+            action_type=ActionType.RUN,
+            reason="required to build 'bar'",
+        ),
+        Action(
+            "foo",
+            Step.BUILD,
+            action_type=ActionType.RUN,
+            reason="organize contents to overlay",
+        ),
+        Action("foo", Step.BUILD, action_type=ActionType.SKIP, reason="already ran"),
+        Action(
+            "foo",
+            Step.STAGE,
+            action_type=ActionType.RUN,
+            reason="required to build 'bar'",
+        ),
+        Action(
+            "bar",
+            Step.BUILD,
+            action_type=ActionType.RERUN,
+            reason="stage for part 'foo' changed",
+        ),
+    ]
+    with lf.action_executor() as ctx:
+        ctx.execute(actions)
+
+    # A request to build all parts skips everything
+    lf = craft_parts.LifecycleManager(parts, **lf_kwargs)
+    actions = lf.plan(Step.BUILD)
+    assert actions == [
+        Action("foo", Step.PULL, action_type=ActionType.SKIP, reason="already ran"),
+        Action("bar", Step.PULL, action_type=ActionType.SKIP, reason="already ran"),
+        Action("foobar", Step.PULL, action_type=ActionType.SKIP, reason="already ran"),
+        Action("foo", Step.OVERLAY, action_type=ActionType.SKIP, reason="already ran"),
+        Action("bar", Step.OVERLAY, action_type=ActionType.SKIP, reason="already ran"),
+        Action(
+            "foobar", Step.OVERLAY, action_type=ActionType.SKIP, reason="already ran"
+        ),
+        Action("foo", Step.BUILD, action_type=ActionType.SKIP, reason="already ran"),
+        Action("bar", Step.BUILD, action_type=ActionType.SKIP, reason="already ran"),
+        Action("foobar", Step.BUILD, action_type=ActionType.SKIP, reason="already ran"),
+    ]
+
+    # Touching a source file triggers an update
+    Path("a.tar.gz").touch()
+    lf = craft_parts.LifecycleManager(parts, **lf_kwargs)
+    actions = lf.plan(Step.BUILD)
+    assert actions == [
+        Action(
+            "foo",
+            Step.PULL,
+            action_type=ActionType.UPDATE,
+            reason="source changed",
+            properties=ActionProperties(changed_files=["a.tar.gz"], changed_dirs=[]),
+        ),
+        Action("bar", Step.PULL, action_type=ActionType.SKIP, reason="already ran"),
+        Action("foobar", Step.PULL, action_type=ActionType.SKIP, reason="already ran"),
+        Action(
+            "foo",
+            step=Step.OVERLAY,
+            action_type=ActionType.UPDATE,
+            reason="'PULL' step changed",
+        ),
+        Action(
+            "foo",
+            step=Step.BUILD,
+            action_type=ActionType.UPDATE,
+            reason="organize contents to overlay",
+        ),
+        Action(
+            "bar", step=Step.OVERLAY, action_type=ActionType.SKIP, reason="already ran"
+        ),
+        Action(
+            "foobar",
+            step=Step.OVERLAY,
+            action_type=ActionType.SKIP,
```

Comment:
I do not think this is due to changes in this PR but I am also surprised to see these 2 actions skipped. I expected at least the `OVERLAY` step for `bar` to be `UPDATE`.